pyro/poutine/handlers.py

```diff
@@ -25,71 +25,131 @@
-    :param trace: a Trace data structure to replay against
-    :param sites: list or dict of names of sample sites in fn to replay against,
-        defaulting to all sites
-    :returns: stochastic function wrapped in a ReplayHandler
-    :rtype: pyro.poutine.ReplayHandler
-
-    Alias for ReplayHandler constructor.
-
     Given a callable that contains Pyro primitive calls,
     return a callable that runs the original, reusing the values at sites in trace
     at those sites in the new trace
+
+    Consider the following Pyro program:
+
+        >>> def model(x):
+        ...     s = pyro.param("s", torch.tensor(0.5))
+        ...     z = pyro.sample("z", dist.Normal(x, s))
+        ...     return z ** 2
+
+    ``replay`` makes ``sample`` statements behave as if they had sampled the values
+    at the corresponding sites in the trace:
+
+        >>> replayed_model = replay(model, trace=old_trace)
+        >>> replayed_model(0.0) == old_trace.nodes["_RETURN"]["value"]
+        True
+
+    :param fn: a stochastic function (callable containing pyro primitive calls)
+    :param trace: a :class:`Trace` data structure to replay against
+    :param sites: list or dict of names of sample sites in fn to replay against,
+        defaulting to all sites
+    :returns: a stochastic function decorated with a :class:`ReplayMessenger`
     """
     msngr = ReplayMessenger(trace=trace, sites=sites)
     return msngr(fn) if fn is not None else msngr
 
 
 def lift(fn=None, prior=None):
     """
-    :param fn: function whose parameters will be lifted to random values
-    :param prior: prior function in the form of a Distribution or a dict of stochastic fns
-    :returns: stochastic function wrapped in LiftHandler
-
     Given a stochastic function with param calls and a prior distribution,
     create a stochastic function where all param calls are replaced by sampling from prior.
     Prior should be a callable or a dict of names to callables.
+
+    Consider the following Pyro program:
+
+        >>> def model(x):
+        ...     s = pyro.param("s", torch.tensor(0.5))
+        ...     z = pyro.sample("z", dist.Normal(x, s))
+        ...     return z ** 2
+        >>> lifted_model = lift(model, prior={"s": dist.Exponential(0.3)})
+
+    ``lift`` makes ``param`` statements behave like ``sample`` statements
+    using the distributions in ``prior``.  In this example, site `s` will now behave
+    as if it was replaced with ``s = pyro.sample("s", dist.Exponential(0.3))``:
+
+        >>> tr = trace(lifted_model).get_trace(0.0)
+        >>> tr.nodes["s"]["type"] == "sample"
+        True
+        >>> tr2 = trace(lifted_model).get_trace(0.0)
+        >>> (tr2.nodes["s"]["value"] == tr.nodes["s"]["value"]).all()
+        False
+
+    :param fn: function whose parameters will be lifted to random values
+    :param prior: prior function in the form of a Distribution or a dict of stochastic fns
+    :returns: ``fn`` decorated with a :class:`LiftMessenger`
```

Comment:
For this to link correctly, I believe you need to use
```
:class:`~pyro.poutine.messengers.LiftMessenger`
```
and similar for other `Messenger` references in this file.